app/main/RTMediaUploadTerms.php

```diff
@@ -0,0 +1,134 @@
+			$general_upload_terms_error_message = $rtmedia->options['general_upload_terms_error_message'];
+
+			if ( ! ( isset( $rtmedia->options ) && isset( $rtmedia->options['styles_enabled'] ) && 0 === $rtmedia->options['styles_enabled'] ) ) {
+				wp_enqueue_style( 'rtmedia-upload-terms-main', RTMEDIA_URL . 'app/assets/css/rtm-upload-terms' . $suffix . '.css', '', RTMEDIA_VERSION );
+			}
+
+			wp_enqueue_script( 'rtmedia-upload-terms-main', RTMEDIA_URL . 'app/assets/js/rtm-upload-terms' . $suffix . '.js', array( 'jquery' ), RTMEDIA_VERSION, true );
+			wp_localize_script( 'rtmedia-upload-terms-main', 'rtmedia_upload_terms_check_terms_message', esc_js( apply_filters( 'rtmedia_upload_terms_check_terms_message', $general_upload_terms_error_message ) ) );
+			wp_localize_script( 'rtmedia-upload-terms-main', 'rtmedia_upload_terms_check_terms_default_message', esc_js( apply_filters( 'rtmedia_upload_terms_check_terms_default_message', __( 'Please check Terms of Service.', 'buddypress-media' ) ) ) );
+		}
+
+		/**
+		 * Render terms and service checkbox in media upload tab.
+		 *
+		 * @global object $rtmedia
+		 * @param  string $content Content after I agree checkbox.
+		 * @return string
+		 */
+		public function show_terms_and_service_checkbox( $content ) {
+			global $rtmedia;
+
+			$options       = $rtmedia->options;
+			$terms_content = '';
+
+			if ( ( ! empty( $options['general_enable_upload_terms'] ) && '0' !== $options['general_enable_upload_terms'] ) && ( ! empty( $options['general_upload_terms_page_link'] ) && '' !== $options['general_upload_terms_page_link'] ) ) {
+				$terms_content = $this->terms_and_service_checkbox_html( $options );
+			}
+
+			return $content . $terms_content;
+		}
+
+		/**
+		 * Adding checkbox on user activity screen.
+		 *
+		 * @param  string $content Content for adding checkbox on user activity screen.
+		 * @return string
+		 */
+		public function show_terms_and_service_checkbox_activity( $content ) {
+			global $rtmedia;
+
+			$options       = $rtmedia->options;
+			$terms_content = '';
+
+			if ( ( ! empty( $options['activity_enable_upload_terms'] ) && '0' !== $options['activity_enable_upload_terms'] ) && ( ! empty( $options['general_upload_terms_page_link'] ) && '' !== $options['general_upload_terms_page_link'] ) ) {
+				$terms_content = $this->terms_and_service_checkbox_html( $options );
+			}
+
+			return $content . $terms_content;
+		}
+
+		/**
+		 * Checkbox of agree terms and condition at front-end.
+		 *
+		 * @param  array  $options Options set from rtMedia settings.
+		 * @return string
+		 */
+		public function terms_and_service_checkbox_html( $options ) {
+			$general_upload_terms_page_link = $options['general_upload_terms_page_link'];
+			$general_upload_terms_message   = $options['general_upload_terms_message'];
+			$content                        = '<div class="rtmedia-upload-terms"> '
```

Comment:
please use a template instead of concatenations
JFI: Golden rule if you need to add more than 2 line of HTML in PHP don't concatenate strings containing HTML, use separate `template file` or use `ob_start()`